src/compiler/checker.ts

```diff
@@ -23895,6 +23895,9 @@ namespace ts {
                     assumeTrue = !assumeTrue;
                 }
                 const valueType = getTypeOfExpression(value);
+                if (assumeTrue && (type.flags & TypeFlags.Unknown) && (operator === SyntaxKind.EqualsEqualsToken) && (valueType.flags & TypeFlags.Null)) {

```

Comment:
I think this should be `(operator === SyntaxKind.EqualsEqualsToken || operator === SyntaxKind.ExclamationEqualsToken)`, so that this narrowing also works if you're using `!=`, just as in the `if` below (from #33488).

Reply:
and if you add this change, don't forget to update the tests :)

Reply:
True that! Will get that in ASAP.

Just a random question but what is `assumeTrue` ... I'll be honest but it was just kind of out of luck where I found using `assumeTrue` helped fix my use case without breaking others however I really don't understand what it means. Would it be cool to explain? 

Thank you!

Reply:
@gabritto I've addressed your comments.

Do you have any tips / advice on how to better understand variables when I come across them? Regarding the `assumeTrue`, I looked in the codebase and it's used in a lot of places but can't track down a comment that explains it's function so I was curious how the Typescript team does it so I can better address issues in the future. 

Thank you :)

Reply:
To be honest, I'm very new to the team and I don't know if there's a methodical way to figure those things out. I don't think there is, sadly. Some things in the code have very descriptive names or comments, but others sometimes don't, as is the case of `assumeTrue`. I'd say if you can't find an explanation by looking at the code, or in the TypeScript wiki, or in https://github.com/orta/typescript-notes, your best bet is to ask someone from the team and we'll be able to help.
@sandersn and @orta might have more tips about where/who to ask those kinds of question.

About `assumeTrue`'s meaning: to give you an example, I figured this one out by looking at the code in `narrowTypeByEquality` and seeing that `assumeTrue` had its boolean value flipped if the condition we had was an inequality (e.g. `a !== b`).
**tl;dr** in `narrowTypeByEquality(type, operator, value, assumeTrue)`, if `assumeTrue` is true we should assume `value` has type `type`. Otherwise, we should assume `value` does *not* have type `type`.
To give you an example:
```ts
var v: unkown;
if (v == null) {
    // ... Here we will call `narrowTypeByEquality` with `assumeTrue` being `true`, because inside this `if` branch, the equality is true, like so: `narrowTypeByEquality(nullType, operator, v, true)`.
}
else {
  // ... Here we will call `narrowTypeByEquality` with `assumeTrue` being `false`, because inside this `else` branch, the equality above is false, like so: `narrowTypeByEquality(nullType, operator, v, false)`.
}
```

Reply:
Aw thank you so much! I really appreciate that insightful explanation 😄 I now better understand what `assumeTrue` means. Super helpful !!